utils/build-script-impl

```diff
@@ -409,58 +443,31 @@ function set_build_options_for_host() {
     local host="$1"
 
     # Hosts which can be cross-compiled must specify:
-    # SWIFT_HOST_TRIPLE and llvm_target_arch (as well as usual HOST_VARIANT flags)
+    # SWIFT_HOST_TRIPLE and llvm_target_arch.
+    # Hosts which have differing platforn names from their
+    # SWIFT_HOST_VARIANT_* values should change them here as well.
+
+    verify_host_is_supported $host
+
+    local IFS='-'
+    local plat_and_arch=($host)
+    local platform=${plat_and_arch[0]}
+    local architecture=${plat_and_arch[1]}
+    local IFS=' '
```

Comment:
Minor: I would prefer `unset IFS`, which should return `IFS` to its previous (global) value or to the default. I would also move it closer to `plat_and_arch`, since I don't think it is necessary for the other two, but it is not that important.